apply/default_applier.go

```diff
@@ -155,7 +156,7 @@ func (c *DefaultApplier) diff() (todoList []ActionName, err error) {
 	todoList = append(todoList, PullIfNotExist)
 	if c.ClusterDesired.Spec.Image != c.ClusterCurrent.Spec.Image {
 		logger.Info("current image is : %s and desired iamge is : %s , so upgrade your cluster", c.ClusterCurrent.Spec.Image, c.ClusterDesired.Spec.Image)
-		todoList = append(todoList, Upgrade)
+		todoList = append(todoList, Upgrade) //TODO check version is same?
```

Comment:
delete the comment, not only version changed need upgrade,but also layer hash changed